firebase-app-distribution/src/main/java/com/google/firebase/app/distribution/AabUpdater.java

```diff
@@ -98,55 +128,36 @@ private void redirectToPlayForAabUpdate(String downloadUrl) {
         safeSetTaskException(
             cachedUpdateTask,
             new FirebaseAppDistributionException(
-                Constants.ErrorMessages.APP_BACKGROUNDED,
-                FirebaseAppDistributionException.Status.DOWNLOAD_FAILURE));
+                Constants.ErrorMessages.APP_BACKGROUNDED, DOWNLOAD_FAILURE));
         return;
       }
     }
 
     // The 302 redirect is obtained here to open the play store directly and avoid opening chrome
-    Executors.newSingleThreadExecutor()
+    executor
         .execute( // Execute the network calls on a background thread
             () -> {
-              HttpsURLConnection connection;
-              String redirect;
+              String redirectUrl;
               try {
-                connection = openHttpsUrlConnection(downloadUrl);
-
-                // To get url to play without redirect we do this connection
-                connection.setInstanceFollowRedirects(false);
-                redirect = connection.getHeaderField("Location");
-                connection.disconnect();
-                connection.getInputStream().close();
-              } catch (FirebaseAppDistributionException | IOException e) {
-                setUpdateTaskCompletionErrorWithDefault(
-                    e,
-                    new FirebaseAppDistributionException(
-                        Constants.ErrorMessages.NETWORK_ERROR,
-                        FirebaseAppDistributionException.Status.DOWNLOAD_FAILURE));
+                redirectUrl = fetchDownloadRedirectUrl(downloadUrl);
+              } catch (FirebaseAppDistributionException e) {
+                setUpdateTaskCompletionError(e);
                 return;
               }
 
-              if (!redirect.isEmpty()) {
-                Intent updateIntent = new Intent(Intent.ACTION_VIEW);
-                updateIntent.setData(Uri.parse(redirect));
-                updateIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-                LogWrapper.getInstance().v(TAG + "Redirecting to play");
-
-                synchronized (updateAabLock) {
-                  lifecycleNotifier.getCurrentActivity().startActivity(updateIntent);
-                  cachedUpdateTask.updateProgress(
-                      UpdateProgress.builder()
-                          .setApkBytesDownloaded(-1)
-                          .setApkFileTotalBytes(-1)
-                          .setUpdateStatus(UpdateStatus.REDIRECTED_TO_PLAY)
-                          .build());
-                }
-              } else {
-                setUpdateTaskCompletionError(
-                    new FirebaseAppDistributionException(
-                        Constants.ErrorMessages.NETWORK_ERROR,
-                        FirebaseAppDistributionException.Status.DOWNLOAD_FAILURE));
+              Intent updateIntent = new Intent(Intent.ACTION_VIEW);
+              updateIntent.setData(Uri.parse(redirectUrl));
+              updateIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
```

Comment:
We might need to do some manual testing here to make sure we don't run into the same updateApp bug. This is different since it's redirecting to play, but the NEW_TASK flag makes me think of it.

Reply:
Ah good point. I added a note to the bug bash doc.